doc/template.html

```diff
@@ -67,5 +67,41 @@ <h3 class="date">$date$</h3>
 $for(include-after)$
 $include-after$
 $endfor$
+<script type="text/javascript">
+  (function() {
+    // write out a basic 'dropdown_link' reference element
+    var dropdown_link = document.createElement('a');
+    dropdown_link.setAttribute('class', 'dropdown_link');
+    dropdown_link.onclick = function(e) {
+    }
+    dropdown_link.innerHTML = ' <b>+</b>'; // the space is for padding
+    dropdown_link.style.cursor = 'pointer';
+    // hack in an ID name for the module list
+    var collapsibleList = document.querySelectorAll('nav ul li > ul > li')
+    collapsibleList.forEach(function(element) {
+      // add a clone of the node to prevent lingering element scoping
+      // otherwise each time a dropdown is clicked it only does the last one
+      var dropdown_link_copy = dropdown_link.cloneNode(true);
+      var list = element.querySelector('ul');
+      if (!list || list.querySelectorAll('li').length == 1)
+        return;
+      list.style.display = 'none';
+      element.setAttribute('state', 'up');
+      dropdown_link_copy.onclick = function() {
```

Comment:
I wonder if it would make more sense to have an `open` and a `close` function that you swap between?..... maybe not.